### PR TITLE
refactor(task-service): remove bpmn code, add abstraction layer

### PR DESCRIPTION
**Pull request by rahulvsf**
## Description

Done the following -

1. Removed unnecessary bpmn related code
2. Added an abstraction layer for internal services
3. Renamed task-example to task-ms-example under sandbox
4. Updated the migration files 

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] Performed a self-review of my own code
- [x] npm test passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the style guide
- [ ] API Documentation in code was updated
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: services/task-service/src/interfaces/event-queue-service.interface.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import {Event} from '../models';
+import {HealthResponse} from '../types';
+
+export interface EventQueueServiceInterface {
+  enqueueEvent(event: Partial<Event>): Promise<void>;
+  healthCheck(): Promise<HealthResponse>;
```
Comment (akshatdubeysf) [inline services/task-service/src/interfaces/event-queue-service.interface.ts:6]: what is the purpose of this healthCheck?

File: services/task-service/src/interfaces/event-queue-service.interface.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import {Event} from '../models';
+import {HealthResponse} from '../types';
+
+export interface EventQueueServiceInterface {
+  enqueueEvent(event: Partial<Event>): Promise<void>;
```
Comment (akshatdubeysf) [inline services/task-service/src/interfaces/event-queue-service.interface.ts:5]: The queue only has a function for adding events in queue? no way to listen for events?

File: services/task-service/src/interfaces/task-operation-service.interface.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+export interface TaskOperationServiceInterface {
+  processTask: (
+    id: string,
+    name: string,
+    key: string,
+    payload?: AnyObject,
+  ) => Promise<void>;
+
+  taskUpdateFlow: (taskKey: string, payload?: AnyObject) => Promise<void>;
```
Comment (akshatdubeysf) [inline services/task-service/src/interfaces/task-operation-service.interface.ts:11]: again what is the purpose of this function? add a description above every interface function if possible

File: services/task-service/src/interfaces/task-operation-service.interface.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import {AnyObject} from '@loopback/repository';
+
+export interface TaskOperationServiceInterface {
+  processTask: (
+    id: string,
+    name: string,
+    key: string,
+    payload?: AnyObject,
```
Comment (akshatdubeysf) [inline services/task-service/src/interfaces/task-operation-service.interface.ts:8]: Why any object? shouldn't this be generic?

File: services/task-service/src/interfaces/utility-service.interface.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import {AnyObject} from '@loopback/repository';
+import {WorkerNameCmdPair} from '@sourceloop/bpmn-service';
+import {ClientAppDTO} from '../models';
+
+export interface UtilityServiceInterface {
+  transformObject(obj?: AnyObject): AnyObject;
```
Comment (akshatdubeysf) [inline services/task-service/src/interfaces/utility-service.interface.ts:6]: again, use generics

File: services/task-service/src/interfaces/utility-service.interface.ts [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+  iterateWorkerMap(
+    workflowName: string,
+    callback: (
+      worker: WorkerNameCmdPair<AnyObject, AnyObject>,
+    ) => Promise<boolean>,
+  ): Promise<boolean>;
+  generateApiKeyAndSecret(clientApp: ClientAppDTO): {
```
Comment (akshatdubeysf) [inline services/task-service/src/interfaces/utility-service.interface.ts:13]: no inline type

File: services/task-service/src/providers/workflow.provider.ts [hunk 7]
```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2023 Sourcefuse Technologies
+//
+// This software is released under the MIT License.
+// https://opensource.org/licenses/MIT
+import {bind, BindingScope, Provider} from '@loopback/core';
+import {HttpErrors} from '@loopback/rest';
+import {WorkflowManagerExtended} from '../types';
+
+@bind({scope: BindingScope.TRANSIENT})
+export class WorkflowProvider implements Provider<WorkflowManagerExtended> {
```
Comment (akshatdubeysf) [inline services/task-service/src/providers/workflow.provider.ts:10]: Why do even need this provider? if it has no actual implementation?

File: services/task-service/src/repositories/event.repository.ts [hunk 8]
```diff
@@ -1,25 +1,15 @@
...
-
-export class EventRepository extends DefaultUserModifyCrudRepository<
+export class EventRepository extends DefaultCrudRepository<
   Event,
   typeof Event.prototype.id
 > {
   constructor(
     @inject(`datasources.${TaskDbSourceName}`)
     dataSource: juggler.DataSource,
-    @inject.getter(AuthenticationBindings.CURRENT_USER)
```
Comment (akshatdubeysf) [inline services/task-service/src/repositories/event.repository.ts:18]: ? how would it getting created by and modified by?

File: services/task-service/src/services/event-queue.service.ts [hunk 9]
```diff
@@ -13,10 +17,10 @@ import {EventProcessorService} from './event-processor.service';
   },
   asLifeCycleObserver,
 )
-export class EventQueueServiceSQS {
+export class EventQueueService implements EventQueueServiceInterface {
   constructor(
     @service(EventProcessorService)
-    private readonly eventProcessorService: EventProcessorService,
+    private readonly eventProcessorService: EventProcessorServiceInterface,
```
Comment (akshatdubeysf) [inline services/task-service/src/services/event-queue.service.ts:23]: This implementation still assumes a particular structure for the connection class

File: services/task-service/src/services/task-operation.service.ts [hunk 10]
```diff
@@ -152,6 +170,19 @@ export class TaskOperationService {
     }
   }
 
+  private async updateTaskStatus(key: string, task: Task) {
+    const dbTask = await this.taskRepo.findOne({
+      where: {
+        key,
```
Comment (akshatdubeysf) [inline services/task-service/src/services/task-operation.service.ts:176]: tenant id?

**Review by akshatdubeysf — CHANGES_REQUESTED** [review]

**Comment by akshatdubeysf** [comment]
@rahulvsf sonar

**Comment by samarpan-b** [comment]
Closing it in favour of #1839

**Pull request closed without merging**